manual/manual/tutorials/coreexamples.etex

```diff
@@ -677,6 +677,36 @@ print_expr e; print_newline ();;
 print_expr (deriv e "x"); print_newline ();;
 \end{caml_example}
 
+There is a "printf" function in "Printf" module that allows you to make
+formatted output more concisely. It models the behavior of the "printf"
+function from the C standard library.
+The "printf" function takes a format string that describes the output values and
+their format, and arguments for output that correspond to the formatting
+directives in the format string.
+\begin{caml_eval}
+open Printf;;
+\end{caml_eval}
+\begin{caml_example}{toplevel}
+printf "%i + %i is an integer value, %F * %F is a float, \"%s\"\n"
+3 2 4.5 1. "this is a string";;
+\end{caml_example}
+The printf function in the OCaml language performs type checking of arguments.
+For example, if you pass it an argument of a type that does not correspond to
+the format specifier, the compiler will display an error message:
+\begin{caml_example}{toplevel}[error]
+printf "Float value: %F" 42;;
+\end{caml_example}
+Associated with this feature is the need to explicitly indicate that the
+string is a format string in order to pass it to the "printf" function.
+\begin{caml_example*}{toplevel}
+let str : ('a, 'b, 'c) format =
+    "%i is an integer value, %F is a float, \"%s\"\n";;
+\end{caml_example*}
```

Comment:
It is probably simpler to use `format_of_string` here. But this is kind of mildly advanced use case, it could be nice to add some warning in the introduction of this paragraph .

Reply:
Yes, `format_of_string` is nice -- it allows to have the string literal checked as a format without having to use an explicit type annotation. If we write an annotation, which may still be useful, we could use `_ format` instead of `('a, 'b, 'c) format`, so that it is clear that we do not give the actual instantiations of type parameters.